### PR TITLE
Reverting backwards compatibility for compilation constants

### PR DESCRIPTION
**Pull request by sfoslund**
Fixes https://github.com/dotnet/sdk/issues/13377

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs [hunk 1]
```diff
@@ -390,72 +388,14 @@ public void It_implicitly_defines_compilation_constants_for_the_target_framework
...
-
         [Theory]
-        [InlineData(new[] { "1.0", "1.1" }, "ios", "1.1", new[] { "IOS", "IOS1_1", "IOS1_0" })]
-        [InlineData(new[] { "11.11", "12.12", "13.13" }, "android", "12.12", new[] { "ANDROID", "ANDROID11_11", "ANDROID12_12" })]
-        [InlineData(new string[] { /* Use the built in SdkSupportedTargetPlatform items */}, "windows", "10.0.19041.0", new[] { "WINDOWS", "WINDOWS7_0", "WINDOWS8_0", "WINDOWS10_0_17763_0", "WINDOWS10_0_18362_0", "WINDOWS10_0_19041_0" })]
-        public void It_implicitly_defines_compilation_constants_for_the_target_platform_with_backwards_compatibility(string[] sdkSupportedTargetPlatform, string targetPlatformIdentifier, string targetPlatformVersion, string[] expectedDefines)
+        [InlineData(new string[] { }, "ios", "1.1", new[] { "IOS", "IOS1_1" })]
+        [InlineData(new string[] { }, "android", "2.2", new[] { "ANDROID", "ANDROID2_2" })]
+        [InlineData(new string[] { }, "windows", "7.0", new[] { "WINDOWS", "WINDOWS7_0" })]
+        [InlineData(new string[] { }, "windows", "10.0.19041.0", new[] { "WINDOWS", "WINDOWS10_0_19041_0" })]
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs:395]: We should not define versioned platform constants: `WINDOWS_10_0_19041`, `ANDROID2_2`, etc.  For now, we should just define the versionless constants: `WINDOWS`, `ANDROID`.

We should look at whether we should have versioned platform constants when we look at adding "or greater" constants for the target framework. @terrajobst

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs [hunk 2]
```diff
@@ -319,6 +316,7 @@ public void It_implicitly_defines_compilation_constants_for_the_configuration(st
         [InlineData("net461", new[] { "NETFRAMEWORK", "NET461" }, true)]
         [InlineData("netcoreapp1.0", new[] { "NETCOREAPP", "NETCOREAPP1_0" }, false)]
         [InlineData("netcoreapp3.0", new[] { "NETCOREAPP", "NETCOREAPP3_0" }, false)]
+        [InlineData("net5.0", new[] { "NETCOREAPP", "NET", "NET5_0" }, false)]
```
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs:319]: What happens when you multi target? Each iteration gets a new set of constants right?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs (hunk 1 above)
Comment (sfoslund) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs]: Sounds good, I've updated to remove these too.

**Review by dsplaisted — APPROVED** [review]

**Review by joeloff — APPROVED** [review]

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs (hunk 2 above)
Comment (sfoslund) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs:319]: Yes, that's right.

**Pull request merged**